pycloudlib/lxd/instance.py

```diff
@@ -318,7 +318,7 @@ def shutdown(self, wait=True, force=False, **kwargs):
 
         subp(cmd)
 
-        if wait:
+        if wait and not self.ephemeral:
```

Comment:
This LGTM, but maybe it would be better to put this login inside `wait_for_stop()` https://github.com/canonical/pycloudlib/blob/0869d07b7894d985e183be957ae8318aae667c00/pycloudlib/lxd/instance.py#L406 just in case there are other places where we call it. Wdyt?

Reply:
@paride your suggestion makes sense. I'm thinking a bit more about this self.ephemeral [relies on calling 'lxc info self.name'](https://github.com/canonical/pycloudlib/tree/main/pycloudlib/lxd/instance.py which may return nothing if the lxc container has already stopped because lxc info will not find the container anymore if it already stopped and was deleted. Since a container won't/can't migrate from ephemeral  TO "not ephemeral". I'm thinking instead that this state could just be preserved on the instance itself at launch time. No need to shell out to lxc info to get that info. I'll adapt this branch with your suggestion and an ephemeral "fix" to see what you think.

Reply:
I think retaining the ephemeral status makes sense, given that it can't change. However it should probably be retained from the `lxc init` stage and not from `lxc launch`, as initialized-but-not-launched instanced already have the ephemeral flag set.

Reply:
On further inspection we can't actually from the "lxc info <name>" logic because that data is not available to [LXD*(Cloud).get_instances](https://github.com/canonical/pycloudlib/tree/main/pycloudlib/lxd/cloud.py#L88-L98). In this case, via a cloud.get_instance call, the instance itself will still have to determine this at runtime.


```bash
$ lxc launch ubuntu-daily:bionic --ephemeral my-ephemeral
$ python3 -c 'import pycloudlib;   lxd = pycloudlib.LXDContainer('test-ephemeral'); instance = lxd.get_instance("my-ephemeral"); print("I am ephemeral: {}".format(instance.ephemeral))'
I am ephemeral True
```
```